services/app-web/src/components/SubmissionSummarySection/RateDetailsSummarySection/RateDetailsSummarySection.tsx

```diff
@@ -174,7 +179,10 @@ export const RateDetailsSummarySection = ({
         }
     }
 
-    useEffect(() => {
+    useDeepCompareEffect(() => {
```

Comment:
was this switch for the flickering? Makes sense that this would be useful but referential equality seems to usually be good enough for us

Reply:
This was switched for infinite re-rendering. I think it is because of `isPreviousSubmission` dependency, which is a hook and the referential equality was not enough.